Fix transparency in skin icon

lumiscosity opened the pull request:
QPainter has a bug where drawing transparency to a freshly initialized, empty QPixmap causes garbage data to be drawn. This broke the rendering of the skin icon. The fix is simply to fill the QPixmap with empty transparent pixels beforehand.

Signed-off-by: lumiscosity <averyrudelphe@gmail.com>

TheKodeToad commented on launcher/SkinUtils.cpp:
```diff
@@ -35,6 +35,7 @@ QPixmap getFaceFromCache(QString username, int height, int width)
         QPixmap skinTexture(fskin.fileName());
         if (!skinTexture.isNull()) {
             QPixmap skin = QPixmap(8, 8);
+            skin.fill(QColor(0,0,0,0));
```
```suggestion
            skin.fill(QColorConstants::Transparent);
```

TheKodeToad commented on launcher/minecraft/auth/MinecraftAccount.cpp:
```diff
@@ -126,6 +126,7 @@ QPixmap MinecraftAccount::getFace() const
         return QPixmap();
     }
     QPixmap skin = QPixmap(8, 8);
+    skin.fill(QColor(0, 0, 0, 0));
```
```suggestion
    skin.fill(QColorConstants::Transparent);
```

TheKodeToad reviewed (changes requested):
Just a suggestion, if you want to keep it the same please format it

lumiscosity commented:
sorry, i wasn't aware such a constant existed! it's been added now

TheKodeToad commented:
~~You forgot to sign off your commits :P~~
Edit: I don't know what that is about

Trial97 commented:
Maybe its because of some GitHub shenanigans  `maple!` vs `lumiscosity` : 
 in the first commit the signoff is with `maple!` but for the rest is `lumiscosity`

From DCO error message:
![image](https://github.com/PrismLauncher/PrismLauncher/assets/42720688/3287a084-da98-40f6-9324-2866389935bf)

lumiscosity commented:
could be; i manually amended the first commit to include the signoff text, and the latter two were accepted with github's "Sign off and commit suggestion" button. sorry, i've never made a pr before

Scrumplex commented:
> sorry, i've never made a pr before

Don't worry about it :D 

It is quite unintuitive honestly

Thanks for your contribution, either way :D

Scrumplex commented:
when taking a look at the log, your first commit was authored by `maple!` while the sign off says `lumiscosity`. The latter should also be `maple!`, so ideally amend it again and change `lumiscosity` to `maple!`

lumiscosity committed:
Fix transparency in skin icon

QPainter has a bug where drawing transparency to a freshly initialized, empty QPixmap causes garbage data to be drawn. This broke the rendering of the skin icon. The fix is simply to fill the QPixmap with empty transparent pixels beforehand.

Signed-off-by: maple! <averyrudelphe@gmail.com>

lumiscosity commented:
alright, i asked a friend who's more knowledgeable about this stuff who told me to just wipe the whole thing and recommit it. should be fine now

TheKodeToad commented:
I think we're waiting for #1486 to be merged before merging any prs targetting develop

Scrumplex committed:
fix: use QColorConstants in Qt 5.14+

Signed-off-by: Sefa Eyeoglu <contact@scrumplex.net>

Scrumplex commented:
this should fix the build on the older Qt 5 version used in one of the Linux builds

Scrumplex commented:
*cough*

TheKodeToad commented:
oh, I forgot we were using older qt versions than that
